[FIX] Improve decomission process

McSim85 opened the pull request:
Often, when I run the role with `state: absent`, the next role run with `state: present` will fail with different errors like described in the bug
https://github.com/artis3n/ansible-role-tailscale/issues/429

After talking to Tailscale support, that might happen if `/var/lib/tailscale/` still contains the previous machine identity (auth key, id, etc). 

We have to remove this folder during decommission.
Fortunately, Ubuntu package has that task inside the `tailscale.postrm`, (if the user rune `apt purge tailscale`):
```sh
$ cat /var/lib/dpkg/info/tailscale.postrm
#!/bin/sh
set -e
if [ -d /run/systemd/system ] ; then
	systemctl --system daemon-reload >/dev/null || true
fi

if [ -x "/usr/bin/deb-systemd-helper" ]; then
    if [ "$1" = "remove" ]; then
		deb-systemd-helper mask 'tailscaled.service' >/dev/null || true
	fi

    if [ "$1" = "purge" ]; then
		deb-systemd-helper purge 'tailscaled.service' >/dev/null || true
		deb-systemd-helper unmask 'tailscaled.service' >/dev/null || true
		rm -rf /var/lib/tailscale
	fi
fi
```
So, this is the best way to handle tailscale state removal - `purge: true,` because this will always be aligned with the package maintainers config.

I am not sure about other distros, though.
Probably, we need a task like this in each distro subtree:
```yaml
- name: Uninstall | Delete Tailscale Package State
  ansible.builtin.file:
    path: "/var/lib/tailscale"
    state: absent
```
@artis3n , please let me know how you prefer to handle this.

artis3n commented on tasks/uninstall.yml:
```diff
@@ -16,6 +16,8 @@
     - tailscale_status.rc != 2
     # "bash: tailscale: command not found"
     - tailscale_status.rc != 127
+    # 500 Internal Server Error: invalid key: API key XXXXX not valid
+    - tailscale_status.rc != 1
```
This is not an acceptable error state, uninstall needs a valid API key

artis3n reviewed (changes requested):
Please remove the invalid API key condition on the de-register command.

Also, please add a verification test to `state-absent` confirming that `/var/lib/tailscale/` no longer exists. Thanks for digging into that with Tailscale and uncovering this!

McSim85 commented:
> Please remove the invalid API key condition on the de-register command.
> 
> Also, please add a verification test to `state-absent` confirming that `/var/lib/tailscale/` no longer exists. Thanks for digging into that with Tailscale and uncovering this!
> 
> Edit: Actually, following https://tailscale.com/kb/1069/uninstall?tab=linux#completely-delete-tailscale , please check for the removal of this state file.

TBH, I think this should be 2 different PRs.
I mean, `purge` fix and `invalid API key` thing.
Because they solve different scopes of problems.
Let me submit a commit with state check and remove "invalid key" and we can focus and discuss this.

McSim85 commented:
Okay, the PR extended.
I introduced a new variable `tailscale_purge_state`.
I go that way because, in some cases, I want to reinstall Tailscale, but save the state.
That is useful for non-ephemeral machines. For example, if I want to save the previous IP. 
Also, I added a separate molecule test to check the state purge scenario.
I am not super experienced with the molecule so I would appreciate a peer review of my changes, @artis3n

artis3n commented on defaults/main.yml:
```diff
@@ -32,3 +32,6 @@ insecurely_log_authkey: false
 auth_key_in_state: true
+# Whether to purge state on uninstall
+# Setting to true will clean up the state folder /var/lib/tailscale
+tailscale_purge_state: false
```
I would like to handle this differently, let's exclude adding a variable in this PR

McSim85 replied:
I will create another PR. 
Because current changes are actively in use :)

McSim85 commented:
I will create another PR.
Because current changes are actively in use :)
